_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw_utils.cpp

```diff
@@ -2657,15 +2657,19 @@ void MfxHwH264Encode::PrepareSeiMessage(
 
 void MfxHwH264Encode::PrepareSeiMessage(
     MfxVideoParam const &   par,
-    mfxExtAvcSeiRecPoint &  msg)
+    mfxExtAvcSeiRecPoint &  msg, 
+    mfxU16 rec_frame_cnt)
 {
     mfxExtCodingOption2 & extOpt2 = GetExtBufferRef(par);
     mfxU32 numTL = par.calcParam.numTemporalLayer;
     if (extOpt2.IntRefType)
         // following calculation assumes that for multiple temporal layers last layer is always non-reference
         msg.recovery_frame_cnt = (extOpt2.IntRefCycleSize - 1) << (numTL > 2 ? (numTL >> 1) : 0);
     else
-        msg.recovery_frame_cnt = par.mfx.GopPicSize;
+        if (par.mfx.GopOptFlag == MFX_GOP_CLOSED)
+            msg.recovery_frame_cnt = 0;
+        else
```

Comment:
Not needed. More safe to use calculated value. Better remove.

Reply:
Done